src/permissions/permissionsManager.ts

```diff
@@ -0,0 +1,124 @@
+import roles from './roles';
+import _get from 'lodash/get';
+import { jwtDecode } from 'jwt-decode';
+
+interface Role {
+  title: string;
+  permissions: string[];
+}
+
+const fallbackRole = 'GUEST';
+
+/**
+ * This function takes a the JWT token of the user and a permission, and validates if the user is allowed to perform the action associated with the permission.
+ * @param {string} token A JWT token string from Firestore User.
+ * @param {string} permission The permission to check.
+ * @returns {Boolean} True if the user has the permission, false otherwise.
+ */
+export function canUser(token: string, permission: string): boolean {
+  const userRole: string = getRoleFromToken(token);
+  const role: Role = _get(roles, userRole.toUpperCase());
+
+  // Check to see if the user's role has the requested permission
+  const roleTitle: string = userRole.toUpperCase();
+  if (roleTitle === 'SUPER_ADMIN') return true;
+  const roleHasPermission = checkPermissionList(role.permissions, permission);
+  if (roleHasPermission) return true;
+
+  // Else, return false
+  return false;
```

Comment:
This could be simplified a little bit and we could remove `lodash`.

The `try/catch` block is important here as `jwtDecode` throws an `InvalidTokenError` in case of malformed JWTs. This need to be caught and handled accordingly (and included in the unhappy path unit tests).

```suggestion
  try {
    const userRole = getRoleFromToken(token)?.toUpperCase()
    
    // If the user is a super admin, grant permission.
    if (userRole === UserRoles.SUPER_ADMIN) return true
    
    const config = roles[userRole as keyof typeof roles]
    
    // If the user role doesn't exist in our config, flag and deny. 
    if (!config) {
      console.error(`[ROAR Permissions Service] Invalid user role "${userRole}".`)
      return false
    }
    
    return checkPermissionList(config.permissions, permission)
  } catch (error) {
    console.error('[ROAR Permissions Service] Error checking permissions:', error)
    return false
  }
```